api/src/main/scala/org/ensime/api/common.scala

```diff
@@ -129,10 +131,80 @@ object OffsetRange extends ((Int, Int) => OffsetRange) {
 // such as files/dirs, existance, content hints
 // (java/scala/class/resource) in the type, validated at construction
 // (and can be revalidated at any time)
-sealed trait EnsimeFile
-final case class RawFile(file: Path) extends EnsimeFile
+sealed trait EnsimeFile {
+  def path: Path
```

Comment:
Hi @fommil, thanks for comments. You are apparently expert in licensing topic, but my observation is that 5-sigma of devs want to code.

Pure data structures cannot do anything useful (besides storing state), for "interesting" stuff we have methods. Or am I missing something? I think `ensimefile.scala` is overusing implicits, therefore I moved it to `EnsimeFile` trait. I believe same approach is used in scala itself, see `List` trait.
What kind of rule do you have in mind?
Is there any reason to use `EnsimeFile` outside of ensime-server?
What "languages" view `EnsimeFile`?
Is there any reason why `util` code uses different license than `api` code?
If some code is deleted from file using GPLv3, does it mean it cannot be added to file which uses Apache2?

Sorry for so many questions, but I always want to learn something new and having discussion is so far the best way of doing that.